x-pack/auditbeat/Jenkinsfile.yml

```diff
@@ -51,31 +56,63 @@ stages:
         mage: "mage build unitTest"
         platforms:             ## override default labels in this specific stage.
             - "windows-2019"
+        when:                  ## Override the top-level when.
+            not_changeset_full_match: "^auditbeat/docs/.*" ## Disable the stage if ONLY DOC changes for the OSS beats
+            branches: true                         ## for all the branches
+            tags: true                             ## for all the tags
```

Comment:
Is there a reason to have this override in `build` and `windows*`, but not in `arm` and `macos`?

Reply:
A brief explanation:

Top-level when condition is the very first evaluation.

Then, any stages with a when condition will be evaluated again. Therefore when for stages and when for top-level don't work together. The initial approach didn't require anything further.

> Is there a reason to have this override in build and windows*, but not in arm and macos?

Mandatory stages with this particular cross-project linting require the `when: not_changeset_full_match...` while `arm` and `macos` are optional stages, so they do need to run on a PR basis unless any of the when conditions.